plugin/web/extensions/sidebar/popups/mobileDropdown/Catalog/index.tsx

```diff
@@ -40,59 +42,132 @@ type Props = {
   setSearchTerm: (searchTerm: string) => void;
+
+  customAddedDatasetDataIDs?: string[];
+  customSearchTerm: string;
+  customExpandedFolders?: {
+    id?: string | undefined;
+    name?: string | undefined;
+  }[];
+  customCatalogProjectName?: string;
+  customCatalogURL?: string;
+  customBackendURL?: string;
+  customBackendProjectName?: string;
+  setCustomExpandedFolders?: React.Dispatch<
+    React.SetStateAction<
+      {
+        id?: string | undefined;
+        name?: string | undefined;
+      }[]
+    >
+  >;
+  onCustomSearch: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  setCustomSearchTerm: (searchTerm: string) => void;
+
+  setSelectedDataset: React.Dispatch<React.SetStateAction<DataCatalogItem | undefined>>;
   onDatasetAdd: (dataset: DataCatalogItem | UserDataItem, keepModalOpen?: boolean) => void;
 };
 
+type Tab = "plateau" | "custom";
+
 const Catalog: React.FC<Props> = ({
-  addedDatasetDataIDs,
   isMobile,
+  inEditor,
+  isCustomProject,
+  selectedDataset,
+
+  addedDatasetDataIDs,
   searchTerm,
   expandedFolders,
-  selectedDataset,
-  inEditor,
   catalogProjectName,
   catalogURL,
   backendURL,
   backendProjectName,
-  setSelectedDataset,
   setExpandedFolders,
   onSearch,
   setSearchTerm,
+
+  customAddedDatasetDataIDs,
+  customSearchTerm,
+  customExpandedFolders,
+  customCatalogProjectName,
+  customCatalogURL,
+  customBackendURL,
+  customBackendProjectName,
+  setCustomExpandedFolders,
+  onCustomSearch,
+  setCustomSearchTerm,
+
+  setSelectedDataset,
   onDatasetAdd,
 }) => {
+  const [currentTab, changeTabs] = useState<Tab>();
   const [filter, setFilter] = useState<GroupBy>("city");
+  const [customFilter, setCustomFilter] = useState<GroupBy>("city");
+
   const [page, setPage] = useState<"catalog" | "details">("catalog");
 
   const [catalogData, setCatalog] = useState<RawDataCatalogItem[]>([]);
+  const [customCatalogData, setCustomCatalog] = useState<RawDataCatalogItem[]>([]);
+
   const [data, setData] = useState<Data[]>();
 
   const processedCatalog = useMemo(() => {
     const c = handleDataCatalogProcessing(catalogData, data);
     return inEditor ? c : c.filter(c => !!c.public);
   }, [catalogData, inEditor, data]);
 
+  const customProcessedCatalog = useMemo(() => {
+    const c = handleDataCatalogProcessing(customCatalogData, data);
+    return inEditor ? c : c.filter(c => !!c.public);
+  }, [customCatalogData, inEditor, data]);
+
   useEffect(() => {
     const catalogBaseUrl = catalogURL || backendURL;
     if (catalogBaseUrl) {
       getDataCatalog(catalogBaseUrl, catalogProjectName).then(res => {
         setCatalog(res);
       });
     }
-  }, [backendURL, catalogProjectName, catalogURL]);
+    const customCatalogBaseUrl = customCatalogURL || customBackendURL;
+    if (customCatalogBaseUrl) {
+      getDataCatalog(customCatalogBaseUrl, customCatalogProjectName).then(res => {
+        setCustomCatalog(res);
+      });
+    }
```

Comment:
This should be in a separate useEffect just so we aren't accidentally recalling both when not needed

Reply:
Updated.